File: 3 КУРС/2 семестр/Учебная практика/JavaProects/app/src/main/java/com/example/voiceassistent/AdviceService.java
package com.example.voiceassistent;

import java.io.IOException;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class AdviceService {
    public static AdviceAPI getAPI(){
        Retrofit retrofit=new Retrofit.Builder()
                .baseUrl("https://api.adviceslip.com/")
                .addConverterFactory(GsonConverterFactory.create().create())
                .build();
        return retrofit.create(AdviceAPI.class);

    }
}

File: 3 КУРС/2 семестр/Учебная практика/JavaProects/app/src/main/java/com/example/voiceassistent/AdviceService.kt
package com.example.voiceassistent

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object AdviceService {
    @JvmStatic
    val aPI: AdviceAPI
        get() {
            val retrofit = Retrofit.Builder()
                    .baseUrl("https://api.adviceslip.com/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
            return retrofit.create(AdviceAPI::class.java)
        }
}